### PR TITLE
intel_adsp: Use device tree to enable/disable each HDA driver

### PR DESCRIPTION
**Pull request by teburd**
Uses the dt_compat_enabled Kconfig preprocessor to set hidden booleans
that are used to toggle the default status of each HDA driver.

Each direction is uniquely selectable which can be useful when building with SOF
where only some direction may wish to be enabled at any given time.

By default, given the device tree (intel_cavs.dtsi) only the host directions are
enabled but an overlay may adjust that as needed.

Alternative to #47524 as it fixes the issue for me.

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

### CODE REVIEW

File: drivers/dma/Kconfig.cavs_hda [hunk 1]
```diff
@@ -3,7 +3,40 @@
...
+	bool "Intel cAVS HDA Link Out DMA drivers"
+	def_bool HAS_DMA_CAVS_HDA_LINK_OUT
+	depends on DMA
+	select DMA_CAVS_HDA
+	help
+	  Intel cAVS Link Out HDA DMA driver.
+
 config DMA_CAVS_HDA
-	bool "Intel cAVS HDA DMA driver"
+	bool "Intel cAVS HDA DMA driver Common Code"
```
[the remark this answers is not in the document]
Comment (teburd) [reply drivers/dma/Kconfig.cavs_hda:39]: Yeah I really wasn't sure of the best route here, I want to build the common code when any of the HDA drivers are enabled, open to suggestions on the best route

File: soc/xtensa/intel_adsp/Kconfig.peripherals [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Intel ADSP Peripheral Options
+#
+# Uses device tree status lines and compitables to determine which drivers
+# to include.
+
+# Copyright (c) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_INTEL_CAVS_HDA_HOST_IN := intel,cavs-hda-host-in
```
[the remark this answers is not in the document]
Comment (teburd) [reply soc/xtensa/intel_adsp/Kconfig.peripherals:9]: Added depends on with a big || setup for the drivers, which seems a little weird to me like a circular ref, but it seems to work

**Comment by teburd** [comment]
@galak thanks for the good suggestions/comments

**Review by dcpleung — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by marc-hb** [comment]
Was this submitted to SOF CI like #47524 was?

**Comment by marc-hb** [comment]
> Was this submitted to SOF CI like https://github.com/zephyrproject-rtos/zephyr/pull/47524 was?

Now it has, I just did. This is progress but it's still not booting on APL UP2:
- https://sof-ci.01.org/sofpr/PR5997/build863/devicetest / https://github.com/thesofproject/sof/pull/5997

```
[   12.360504] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: ------------[ DSP dump start ]------------
[   12.360516] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: Firmware boot failure due to timeout
[   12.360521] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: fw_state: SOF_FW_BOOT_IN_PROGRESS (2)
[   12.360532] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: 0x00000005: module: ROM, state: FW_ENTERED, running
[   12.360559] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: invalid header size 0x1010202. FW oops is bogus
[   12.360564] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: unexpected fault 0x00000000 trace 0x00000000
[   12.360571] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: ------------[ DSP dump end ]------------
[   12.360575] kernel: sof-audio-pci-intel-apl 0000:00:0e.0: error: failed to boot DSP firmware -5
[   12.360584] kernel: snd_sof:sof_set_fw_state: sof-audio-pci-intel-apl 0000:00:0e.0: fw_state change: 2 -> 3
[   12.361614] kernel: snd_sof_intel_hda_common:hda_dsp_core_reset_enter: sof-audio-pci-intel-apl 0000:00:0e.0: FW Poll Status: reg[0x4]=0x1010303 successful
[   12.361634] kernel: snd_sof_intel_hda_common:hda_dsp_core_power_down: sof-audio-pci-intel-apl 0000:00:0e.0: FW Poll Status: reg[0x4]=0x303 successful
[   12.361644] kernel: snd_sof_intel_hda_common:hda_dsp_core_is_enabled: sof-audio-pci-intel-apl 0000:00:0e.0: DSP core(s) enabled? 0 : core_mask 3
[   12.362320] kernel: snd_sof:sof_set_fw_state: sof-audio-pci-intel-apl 0000:00:0e.0: fw_state change: 3 -> 0
[   12.362502] kernel: sof-audio-pci-intel-apl: probe of 0000:00:0e.0 failed with error -5
```

I updated my revert and it still boots and even passes all tests:
- https://sof-ci.01.org/sofpr/PR5996/build864/devicetest / https://github.com/thesofproject/sof/pull/5996

**Pull request merged**